aloha-vw-jni/src/test/scala/com/eharmony/aloha/models/vw/jni/multilabel/VwMultilabelDownsampledModelTest.scala

```diff
@@ -0,0 +1,188 @@
+package com.eharmony.aloha.models.vw.jni.multilabel
+
+import java.io.File
+
+import com.eharmony.aloha.audit.impl.OptionAuditor
+import com.eharmony.aloha.dataset.vw.multilabel.VwDownsampledMultilabelRowCreator
+import com.eharmony.aloha.dataset.vw.multilabel.json.VwDownsampledMultilabeledJson
+import com.eharmony.aloha.factory.ModelFactory
+import com.eharmony.aloha.id.ModelId
+import com.eharmony.aloha.io.StringReadable
+import com.eharmony.aloha.io.vfs.Vfs
+import com.eharmony.aloha.semantics.compiled.CompiledSemanticsInstances
+import org.apache.commons.vfs2.VFS
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.BlockJUnit4ClassRunner
+import org.junit.Assert.{assertEquals, fail}
+import spray.json.pimpString
+import spray.json.DefaultJsonProtocol.IntJsonFormat
+import vowpalWabbit.learner.{VWActionScoresLearner, VWLearners}
+
+import scala.collection.breakOut
+import scala.util.Random
+
+/**
+  * Created by ryan.deak on 11/6/17.
+  */
+@RunWith(classOf[BlockJUnit4ClassRunner])
+class VwMultilabelDownsampledModelTest {
+
+  import VwMultilabelDownsampledModelTest.EndToEndDatasetSpec
+
+  @Test def test(): Unit = {
```

Comment:
I don't see a test for the actual VW input.  Is that here?

Reply:
Yeah.  Sort of.  The VW input is generated implicitly and the prediction output is produced and tested against the marginal probabilities.  So, it works implicitly.